atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java

```diff
@@ -354,9 +355,15 @@ private HikariDataSource getDataSourcePool() {
                 }
             }
         } catch (PoolInitializationException e) {
+            Optional<Integer> maybeSqlErrorCode = Optional.ofNullable(e.getCause())
```

Comment:
It's not uncommon (at least for Oracle) for SQLExceptions to be nested multiple levels deep and the top most code may not be all the informative. 

Something like below has a better chance of helping identify the actual problem:

        for (Throwable throwable : exception) {
            if (throwable instanceof SQLException) {
                SQLException sqlException = (SQLException) throwable;
                log.error(
                        "Failed to initialize hikari data source",
                        SafeArg.of("errorCode", sqlException.getErrorCode()),
                        SafeArg.of("errorState", sqlException.getSQLState()),
                        UnsafeArg.of("message", sqlException.getMessage()),
 UnsafeArg.of("url", connConfig.getUrl())
                        sqlException);
            }